flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/AggregationMergeFunction.java

```
@@ -0,0 +1,104 @@
+ * non-null fields on merge.
+ */
+public class AggregationMergeFunction implements MergeFunction {
+
+    private static final long serialVersionUID = 1L;
+
+    private final RowData.FieldGetter[] getters;
+
+    private transient GenericRowData row;
+
+    public AggregationMergeFunction(RowData.FieldGetter[] getters) {
+        this.getters = getters;
+    }
+
+    @Override
+    public void reset() {
+        this.row = new GenericRowData(getters.length);
+    }
+
+    @Override
+    public void add(RowData value) {
+        for (int i = 0; i < getters.length; i++)
+        {
+            Object currentField = getters[i].getFieldOrNull(value);
+            Object oldValue = row.getField(i);
+            Object result = sum(oldValue, currentField);
+            if (result != null)
+            {
+                row.setField(i, result);
+            }
+        }
+    }
+
+    private Object sum(Object oldValue, Object currentField) {
+        if (currentField == null)
+        {
+            return null;
+        }
+        if (oldValue == null)
+        {
+            return currentField;
+        }
+        if (oldValue instanceof Integer && currentField instanceof Integer)
+        {
+            return Integer.sum((Integer) oldValue, (Integer) currentField);
+        }
+        else if (oldValue instanceof Long && currentField instanceof Long)
+        {
+            return Long.sum((Long) oldValue, (Long) currentField);
+        }
+        else if (oldValue instanceof Double && currentField instanceof Double)
+        {
+            return Double.sum((Double) oldValue, (Double) currentField);
+        }
+        else if (oldValue instanceof Float && currentField instanceof Float)
+        {
+            return Float.sum((Float) oldValue, (Float) currentField);
+        }
+        else if (oldValue instanceof String && currentField instanceof String)
+        {
+            return "null";
+        }
+        return null;
```

Throws exception instead of returning null for types which do not support summing. It is better to explicitly tell the user what we do not support instead of quietly do something they might not expect. Also it would be better to perform this check in the constructor.

---

Is there a more explicit exception than RuntimeException?